src/sentry/integrations/api/serializers/rest_framework/data_forwarder.py

```diff
@@ -0,0 +1,206 @@
+import re
+from collections.abc import MutableMapping
+from typing import Any
+
+from rest_framework import serializers
+from rest_framework.serializers import Serializer, ValidationError
+
+from sentry.integrations.models.data_forwarder import DataForwarder
+from sentry.integrations.models.data_forwarder_project import DataForwarderProject
+from sentry.models.organization import Organization
+from sentry.models.project import Project
+
+
+class DataForwarderSerializer(Serializer):
+    organization_id = serializers.IntegerField()
+    is_enabled = serializers.BooleanField(default=True)
+    enroll_new_projects = serializers.BooleanField(default=False)
+    provider = serializers.ChoiceField(
+        choices=[
+            ("segment", "Segment"),
+            ("sqs", "Amazon SQS"),
+            ("splunk", "Splunk"),
+        ]
+    )
+    config = serializers.JSONField(default=dict)
+
+    def validate_organization_id(self, value: int) -> int:
+        try:
+            Organization.objects.get(id=value)
+        except Organization.DoesNotExist:
+            raise ValidationError("Organization with this ID does not exist")
+        return value
+
+    def validate_config(self, value: dict) -> dict:
+        provider = self.initial_data.get("provider") or getattr(self.instance, "provider", None)
+
+        if provider == "sqs":
+            return self._validate_sqs_config(value)
+        elif provider == "segment":
+            return self._validate_segment_config(value)
+        elif provider == "splunk":
+            return self._validate_splunk_config(value)
+
+        return value
+
+    def _validate_sqs_config(self, config: dict) -> dict:
```

Comment:
Could we break up this function a bit more into some helpers? It looks like there a few sqs validation steps nested here?